src/config.cpp

```diff
@@ -59,6 +66,106 @@ void parse_uint32(void* value_ptr, const char* txt) {
     cfgval->set = true;
 }
 
+// Helper function to trim whitespace and remove quotations
+std::string trim(const std::string& str) {
+    size_t first = str.find_first_not_of(" \t\n\""); // Also include quotes
+    size_t last = str.find_last_not_of(" \t\n\"");
+
+    // Check for valid range
+    if (first == std::string::npos || last == std::string::npos) {
+        return str; // Empty or only whitespace/quotes
+    }
+
+    // Extract the trimmed and unquoted substring
+    std::string trimmed = str.substr(first, (last - first + 1));
+
+    // Check if the trimmed string still starts and ends with quotes
+    if (trimmed.front() == '"' && trimmed.back() == '"') {
+        return trimmed.substr(1, trimmed.size() - 2); // Remove the quotes
+    } else {
+        return trimmed;
+    }
+}
+
+void read_config_file(std::unordered_map<std::string, std::string>& umap,
+                      std::ifstream& config_stream) {
+
+    std::string line;
+    while (std::getline(config_stream, line)) {
+        // Ignore comments and empty lines
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+        line = trim(line);
+        // Check for section headers
+        if (line[0] == '[' && line[line.size() - 1] == ']') {
+            continue;
```

Comment:
Why do we have any support for section headers? These are not semantic in any way and in effect are just an alternative comment syntax :). If you have plans for them in the future, fine to keep this in, otherwise I'd suggest removing this (more untested code!).

Reply:
Just removed it for now.